backend/libs/watcher-operations.spec.ts

```diff
@@ -0,0 +1,275 @@
+            );
+        });
+    });
+
+    describe('when the room is in a playing state (synchronized start was initiated previously)', () => {
+        beforeEach(async (done) => {
+            room!.videoStatus = VideoSyncStatus.PLAYING;
+            room!.currentVideoUrl = 'https://www.youtube.com/watch?v=4242';
+            await updateRoom(room!, tableName, ddb);
+            done();
+        });
+        describe('when receiving a play event from the owner', () => {
+            beforeEach(async (done) => {
+                playerEvent = {
+                    mediaEventType: MediaEventType.PLAY,
+                    duration: 42,
+                    currentTime: 42,
+                    now: new Date(),
+                };
+                await updateWatcherVideoStatus(
+                    room!,
+                    tableName,
+                    'owner',
+                    playerEvent,
+                    ddb,
+                );
+                done();
+            });
+            it('updates the watcher to the PLAYING state', async () => {
+                const { Item } = await ddb
+                    .get({
+                        TableName: tableName,
+                        Key: { roomId },
+                    })
+                    .promise();
+                if (!Item) {
+                    throw new Error('Error retrieving DDB room');
+                }
+
+                expect(Item.watchers['owner']).toEqual({
+                    connectionId: 'owner',
+                    currentVideoStatus: UserVideoStatus.PLAYING,
+                    id: 'owner',
+                    initialSync: false,
+                    userAgent: 'TODO',
+                });
+            });
+        });
+        describe('when receiving a pause event from the owner', () => {
+            beforeEach(async (done) => {
+                playerEvent = {
+                    mediaEventType: MediaEventType.PAUSE,
+                    duration: 42,
+                    currentTime: 42,
+                    now: new Date(),
+                };
+                await updateWatcherVideoStatus(
+                    room!,
+                    tableName,
+                    'owner',
+                    playerEvent,
+                    ddb,
+                );
+                done();
+            });
+            it('updates the watcher to the BUFFERING state', async () => {
+                const { Item } = await ddb
+                    .get({
+                        TableName: tableName,
+                        Key: { roomId },
+                    })
+                    .promise();
+                if (!Item) {
+                    throw new Error('Error retrieving DDB room');
+                }
+
+                expect(Item.watchers['owner']).toEqual({
+                    connectionId: 'owner',
+                    currentVideoStatus: UserVideoStatus.BUFFERING,
```

Comment:
It may not be the right place but I don't like to be buffering. We don't know if it's true. We should use WAITING instead no ?

Reply:
hmm I had picked buffering because it felt more accurate than waiting. Apart from buffering, you're either in an unknown state (you never tried to play the video) or ready to start the video because you've already attempted to start it. Maybe we ca discuss the terminology with the team.

Reply:
Ok